gensim/models/lsimodel.py

```diff
@@ -489,7 +489,8 @@ def add_documents(self, corpus, chunksize=None, decay=None):
             chunksize = self.chunksize
         if decay is None:
             decay = self.decay
-
+        if corpus == []:
```

Comment:
```suggestion
        if not corpus:
```

Reply:
Perhaps "if len(corpus) < 0" - if not corpus would also catch corpus = False etc (I know, unlikely, but would be very confusing), and corpus=True would pass no issue. Also I think corpus can very feasibly be None which would have the same issue

Reply:
All these cases ought to result in the same warning IMO.